### ISSUE TITLE
Mounting of IAM roles for service accounts web identity token fails

### ISSUE THREAD

**Issue by cyrus-mc** [telepresenceio/telepresence#3166]
**Describe the bug**

I have a deployment that uses [IAM roles for service accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). When a workload is configured for IAM roles a web identity token is mounted into the POD at location /var/run/secrets/eks.amazonaws.com.

When intercepting this type of workload I would have expected that this mount point would be reflected in the exported mounts such that local workloads could assume the identify assigned to the POD. However this does not seem to be the case as the only mount exported is /var/run/secrets/kubernetes.io as can be seen from the logs of the `traffic-agent`:

```
2023-05-08 18:48:54.1240 info    Enhanced Traffic Agent v1.13.13
2023-05-08 18:48:54.1244 debug   Logging at this level "debug"
2023-05-08 18:48:54.1244 info    Adding exported mounts for container node
2023-05-08 18:48:54.1245 debug   checking agent secrets mount path: /var/run/secrets/kubernetes.io
2023-05-08 18:48:54.1245 debug   checking appmounts directory: /var/run/secrets/kubernetes.io
2023-05-08 18:48:54.1245 debug   create appmounts directory: /tel_app_exports/node/var/run/secrets/kubernetes.io
2023-05-08 18:48:54.1246 debug   create appmounts symlink: /var/run/secrets/kubernetes.io /tel_app_exports/node/var/run/secrets/kubernetes.io
2023-05-08 18:48:54.1246 info    new agent secrets mount path: /var/run/secrets/kubernetes.io
```

**To Reproduce**

This problem is fairly simple to reproduce. Simply deploy an application configured for IAM roles for service accounts and intercept that service specifying `--mount=true`. Then view the TELEPRESENCE_ROOT directory and verify that you do not see the directory `/var/run/secrets/eks.amazonaws.com` mounted.

**Expected behavior**

I would expect telepresence to export all mounts found in /var/run/secrets as documented in the CHANGELOG:

```
### 2.4.1 (August 30, 2021)

- Bugfix: Telepresence will now mount all directories from `/var/run/secrets`, not just the kubernetes.io ones.
  This allows the mounting of secrets directories such as eks.amazonaws.com (for IRSA tokens)
```

**Comment by cyrus-mc** [comment]
I believe the reason for this is the `traffic-agent` container does not see the mount point of `/var/run/secrets/eks.amazonaws.com` that is mounted in the primary workload container. 

IAM roles for service accounts is accomplished through its own mutatingwebhook in which the POD is mutated to add the `token` volume. What I suspect is happening is an ordering thing. The most likely cause is that the IAM roles for service accounts `mutatingwebhook` fires prior to the traffic agent mutation and therefore does not get the additional `token` volume injected.

**Comment by cyrus-mc** [comment]
@thallgren 

Looking at the code I believe this is the relevant section that is responsible for looping over the containers and adding the volume mounts to the traffic-agent:

https://github.com/telepresenceio/telepresence/blob/release/v2/pkg/agentconfig/container.go#L250

I may be reading this wrong but line 273 says `only add /var/run/secrets once, not once per container`. And the code looks at the mount path and if it is prefixed with `/var/run/secrets` it just continues rather than add it to the mount list.

This might explain why the mount for the `eks.amazonaws.com` token is skipped. However I tested out my theory by creating a `emptyDir` volumeMount at location `/var/run/secrets/testing` and that showed up when I exported mounts (and was in the traffic-agent container) so I am not sure I am interpreting that logic correctly.
